### PR TITLE
Audit and update App Components docs

### PR DESCRIPTION
**Pull request by aw-asana**
Includes content on the new app gallery. 

Other updates include updating old screenshots (e.g., new developer console UI), an audit of content, etc.

### CODE REVIEW

File: source/includes/markdown/_25-app-components.html.md [hunk 1]
```diff
@@ -309,148 +308,149 @@ https://app-server.com/rule?workspace=12345&project=23456&action_type=45678&acti
...
 
-### Start Building the App Server
+### Start building the app server
 
-Because App Components allow your end users to interact with resources outside of Asana's user interface, an [App Server](/docs/app-server) is required for building with App Components. While it isn't necessary to have a complete App Server before moving forward, it's a good idea to at least know the URLs to any server endpoints you want to configure ahead of time (e.g., your URL to [get the metadata](/docs/get-widget-metadata) for a [Widget](/docs/widget)), though these can be configured or changed at any time.
+Because app components allow your end users to interact with resources outside of Asana's user interface, an [app server](/docs/app-server) is required for building with app components. While it isn't necessary to have a complete app server before moving forward, it's a good idea to at least know the URLs to any server endpoints you want to configure ahead of time (e.g., your URL to [get the metadata](/docs/get-widget-metadata) for a [sidget](/docs/widget)), though these can be configured or changed at any time.
```
Comment (jv-asana) [inline source/includes/markdown/_25-app-components.html.md:406]: I think there's a spelling mistake `sidget`

File: source/includes/markdown/_25-app-components.html.md [hunk 2]
```diff
@@ -459,20 +459,20 @@ Follow the links in the table below to view the configurations required to build
...
 | [Lookup](/docs/lookup)            | Show a text input to allow users to find and attach resources to tasks. Users can paste a URL, ID, or pick from an optional typeahead.                  |
-| [Rule Actions](/docs/rule-action) | Build one or more custom action for Asana’s Rules engine to help users automate their work. Users can create rules that run your action when triggered. |
-| [Entry Point](/docs/entry-point)  | Configure the button in tasks that initiates the Lookup and Modal Form. To configure this, one or both of these capabilities must be configured first.  |
+| [Rule actions](/docs/rule-action) | Build one or more custom action for Asana’s [Rules](https://asana.com/guide/help/premium/rules) engine to help users automate their work. Users can create rules that run your action when triggered. |
+| [Entry point](/docs/entry-point)  | Configure the button in tasks that initiates the lookup and dodal Form. To configure this, one or both of these capabilities must be configured first.  |
```
Comment (jv-asana) [inline source/includes/markdown/_25-app-components.html.md:465]: spelling mistake `dodal Form` I think you meant `modal form`?

File: source/includes/markdown/_25-app-components.html.md (hunk 1 above)
Comment (aw-asana) [reply source/includes/markdown/_25-app-components.html.md:406]: Fixed!

File: source/includes/markdown/_25-app-components.html.md (hunk 2 above)
Comment (aw-asana) [reply source/includes/markdown/_25-app-components.html.md:465]: Fixed!

**Review by jv-asana — APPROVED** [review]
LGTM

**Pull request merged**